OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs

```diff
@@ -20,24 +22,26 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("Ability")]
 	public class CaptureProperties : ScriptActorProperties
 	{
-		readonly CapturesInfo normalInfo;
+		readonly Captures[] capturesTraits;
 		readonly ExternalCapturesInfo externalInfo;
 
 		public CaptureProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			normalInfo = Self.Info.TraitInfoOrDefault<CapturesInfo>();
+			capturesTraits = Self.TraitsImplementing<Captures>().ToArray();
 			externalInfo = Self.Info.TraitInfoOrDefault<ExternalCapturesInfo>();
 		}
 
 		[Desc("Captures the target actor.")]
 		public void Capture(Actor target)
 		{
+			var captures = capturesTraits.FirstOrDefault(x => !x.IsTraitDisabled && x.Info.CaptureTypes.Overlaps(target.GetEnabledTargetTypes()));
+
 			var normalCapturable = target.Info.TraitInfoOrDefault<CapturableInfo>();
 			var externalCapturable = target.Info.TraitInfoOrDefault<ExternalCapturableInfo>();
 
-			if (normalInfo != null && normalCapturable != null && normalInfo.CaptureTypes.Contains(normalCapturable.Type))
-				Self.QueueActivity(new CaptureActor(Self, target));
+			if (captures != null && capturesTraits != null && normalCapturable != null)
```

Comment:
You don't need to check for capturesTraits here, that can't be null. (It would be an empty array at worst and even in that case, captures will be null so this check doesn't get evaulated.)